src/Features/LanguageServer/Protocol/Handler/Completion/CompletionHandler.cs

```diff
@@ -77,22 +79,51 @@ internal class CompletionHandler : IRequestHandler<LSP.CompletionParams, LSP.Com
             var completionTrigger = await ProtocolConversions.LSPToRoslynCompletionTriggerAsync(request.Context, document, position, cancellationToken).ConfigureAwait(false);
 
             var list = await completionService.GetCompletionsAsync(document, position, completionTrigger, options: completionOptions, cancellationToken: cancellationToken).ConfigureAwait(false);
-            if (list == null)
+            if (list == null || list.Items.IsEmpty)
             {
                 return null;
             }
 
             var lspVSClientCapability = context.ClientCapabilities?.HasVisualStudioLspCapability() == true;
-
             var commitCharactersRuleCache = new Dictionary<ImmutableArray<CharacterSetModificationRule>, ImmutableArray<string>>();
 
             // Cache the completion list so we can avoid recomputation in the resolve handler
             var resultId = await _completionListCache.UpdateCacheAsync(list, cancellationToken).ConfigureAwait(false);
 
+            // Feature flag to enable the return of TextEdits instead of InsertTexts (will increase payload size).
+            // Flag is defined in VisualStudio\Core\Def\PackageRegistration.pkgdef.
+            var featureFlagService = context.Solution.Workspace.Services.GetRequiredService<IExperimentationService>();
+            var returnTextEdits = featureFlagService.IsExperimentEnabled("Roslyn.LSP.Completion");
+
+            SourceText? documentText = null;
```

Comment:
don't really need this value.

Reply:
We need it to potentially re-compute the Range [later on in the file](https://github.com/dotnet/roslyn/blob/5368354063ca6855664f09eadfc9d1bf3763d537/src/Features/LanguageServer/Protocol/Handler/Completion/CompletionHandler.cs#L257). Storing the text allows us to avoid asking for the text again.